lib/assignDisabledRanges.js

```diff
@@ -66,38 +85,43 @@ module.exports = function(
 
     if (ruleName === ALL_RULES) {
       Object.keys(disabledRanges).forEach(disabledRuleName => {
-        startDisabledRange(line, disabledRuleName);
-        endDisabledRange(line, disabledRuleName);
+        const strict = disabledRuleName === ALL_RULES;
+
+        startDisabledRange(line, disabledRuleName, strict);
+        endDisabledRange(line, disabledRuleName, strict);
       });
     } else {
-      startDisabledRange(line, ruleName);
-      endDisabledRange(line, ruleName);
+      startDisabledRange(line, ruleName, true);
+      endDisabledRange(line, ruleName, true);
```

Comment:
provided `ruleName` is not `all`, e.g.:
`/* stylelint-disable my-rule */`
so `strictStart`/`strictEnd` true